### PR TITLE
Feat/int 778 add a diff viewer in the webview

### PR DESCRIPTION
**Pull request by henoktsegayetoptal**

### CODE REVIEW

File: intuita-webview/public/index.html [hunk 1]
```diff
@@ -8,7 +8,11 @@
 			name="description"
 			content="Web site created using create-react-app"
 		/>
-		<title>Hello World</title>
+		<meta
+			http-equiv="Content-Security-Policy"
+			content="default-src 'self'; style-src 'self' 'unsafe-inline'"
```
Comment (grzpab) [inline intuita-webview/public/index.html:13]: why was this necessary?

File: src/extension.ts [hunk 2]
```diff
@@ -162,6 +163,34 @@ export async function activate(context: vscode.ExtensionContext) {
 	const intuitaTextDocumentContentProvider =
 		new IntuitaTextDocumentContentProvider();
 
+	context.subscriptions.push(
+		vscode.commands.registerCommand(
+			'intuita.openJobDiff',
+			async (jobHash?: JobHash) => {
+				if (!jobHash || !rootPath) return;
```
Comment (grzpab) [inline src/extension.ts:170]: ` { return; }`

File: intuita-webview/src/jobDiffView/DiffViewer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	const containerRef = useRef<HTMLDivElement>(null);
+	const { width: containerWidth } =
+		useElementSize<HTMLDivElement>(containerRef);
+	const [viewType, setViewType] = useState<'inline' | 'side-by-side'>(() => {
+		return [
+			JobKind.copyFile,
+			JobKind.moveFile,
+			JobKind.deleteFile,
+			JobKind.createFile,
+		].includes(jobKind as unknown as JobKind)
```
Comment (grzpab) [inline intuita-webview/src/jobDiffView/DiffViewer/index.tsx:34]: I have a feeling I already saw this statement, can you make it a utility function?

File: intuita-webview/src/JobDiffView/app.css [hunk 4]
```diff
@@ -0,0 +1,52 @@
+.flex {
```
[the remark this answers is not in the document]
Comment (henoktsegayetoptal) [reply intuita-webview/src/JobDiffView/app.css:1]: created those style

**Review by hbjORbj — COMMENTED** [review]
Pressing "Copy" Job doesn't display anything. I think it should be tested with all job kinds? Also, if you were viewing diff view of "Rewrite" job and then press "Copy" job, the diff view of "Rewrite" job is still there being displayed.

File: intuita-webview/src/jobDiffView/DiffViewer/index.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
+import { useRef, useState } from 'react';
+import ReactDiffViewer from 'react-diff-viewer-continued';
+import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
+import { useElementSize } from '../hooks/useElementSize';
+import { Container } from './Container';
+import { JobDiffViewProps } from '../App';
+
+enum JobKind {
```
[the remark this answers is not in the document]
Comment (henoktsegayetoptal) [reply intuita-webview/src/jobDiffView/DiffViewer/index.tsx:8]: Yes

**Comment by henoktsegayetoptal** [comment]
> Pressing "Copy" Job doesn't display anything. I think it should be tested with all job kinds? Also, if you were viewing diff view of "Rewrite" job and then press "Copy" job, the diff view of "Rewrite" job is still there being displayed.

@hbjORbj  this is because the copy job was not working in general. the copy was not working in main. but since its fixed it should work

File: intuita-webview/src/jobDiffView/hooks/useElementSize.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+				height: element.offsetHeight,
+			});
+		}
+
+		if (element) {
+			handleResize();
+			window.addEventListener('resize', handleResize);
+		}
+		return () => {
+			window.removeEventListener('resize', handleResize);
```
Comment (grzpab) [inline intuita-webview/src/jobDiffView/hooks/useElementSize.ts:28]: what happens if the event listener was not set because the element was missing?

File: intuita-webview/src/JobDiffView/index.css [hunk 7]
```diff
@@ -0,0 +1,55 @@
+.flex {
```
[the remark this answers is not in the document]
Comment (henoktsegayetoptal) [reply intuita-webview/src/JobDiffView/index.css:1]: I created it

File: intuita-webview/public/index.html (hunk 1 above)
Comment (henoktsegayetoptal) [reply intuita-webview/public/index.html]: its not

**Comment by henoktsegayetoptal** [comment]
https://github.com/intuita-inc/intuita-vscode-extension/pull/245
